### PR TITLE
Generalise to other packages

### PR DESCRIPTION
**Pull request by seabbs**
* Added a wrapper for `bsts` models that takes the old internals of `forecast_rt`. The idea is to put in a vector `y`, samples, and a time horizon and to get out a data.frame of samples (rows) by horizon (columns). 
* Updated all examples to work with the new wrapper
* Updated docs to match change.
* Stop raw returns form `evaluate_model` being the default
* Add `bsts_model` wrapper for `fable` (`fable_model`).
* Add other models to docs and examples.
* Added support for uncertain observations plotting (via reducing alpha and plotting observations as points).
* Provide docs on how to add generic models.

## Comment 

`fable` has several dependencies for different models that are required in order for it to run them. The error checking in `compare_timeseries`  screens out these messages which can make it hard to debug. The model implementation process should include an initial check that the `fable` model works at the lowest level (i.e. with `fable_model` as per the examples). Similarly `fable` natively uses the `future` package for parallelisation but also requires the `future.apply` package which is not installed by default. If a `future` is detected (i.e. for use with `compare_timeseries`) then  `future.apply` must be installed in order for a `fable` model to be run, even though it is not running in parallel in this instance.  Obviously these features are not ideal and quite annoying on initial set up if not aware. Some flagging has been added to the `fable_model` docs to indicate the problem.

### CODE REVIEW

File: R/bsts_model.R [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#'
+#' ## Used on its own
+#' bsts_model(y = EpiSoon::example_obs_rts[1:10, ]$rt,
+#'            model = function(ss, y){
+#'            bsts::AddAr(ss, y = y, lags = 2)},
+#'            samples = 10, horizon = 7)
+#'
+#' ## Used for forecasting
+#'  forecast_rt(EpiSoon::example_obs_rts[1:10, ],
+#'                     model = function(...){EpiSoon::bsts_model(model =
```
Comment (nikosbosse) [inline R/bsts_model.R:20]: the nested function looks like quite an ingenious hack, but somehow still less then ideal. 
Also at line 28 you lose the flexibility to add something to ss, if you force it to pass in list()
don't think this is worth working on right now, just wanted to flag it.

Comment (seabbs) [reply R/bsts_model.R:20]: Keen to see an alternative implementation?

Comment (seabbs) [reply R/bsts_model.R:20]: Perhaps open an issue outlining the current limitations and we can review if a solution is immediatly obvious

File: R/fable_model.R [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#'forecast_rt(EpiSoon::example_obs_rts[1:10, ],
+#'            model = function(...){
+#'            fable_model(model = fable::ARIMA(y ~ time), ...)},
+#'            horizon = 7, samples = 10)
+fable_model <- function(y = NULL, samples = NULL,
+                       horizon = NULL, model = NULL) {
+
+
+## Make input numeric into correct tsibble format
+timeseries <- tsibble::tsibble(y = y, time = 1:length(y), index = time)
```
Comment (nikosbosse) [inline R/fable_model.R:34]: we are passing in a date column anyway, right? I had used as_tsibble(index = date, key = timeseries), to preserve the given dates. The key = timeseries argument is just omitted I guess for our purposes

File: R/forecast_cases.R [hunk 3]
```diff
@@ -18,8 +18,9 @@
 #' @examples
 #' ## Rt forecast
 #' forecast <- forecast_rt(EpiSoon::example_obs_rts[1:10, ],
-#'                         model = function(ss, y) {
-#'                         bsts::AddAutoAr(ss, y = y, lags = 7)
+#'                         model = function(...){
+#'                         EpiSoon::bsts_model(model = function(ss, y){
+#'                         bsts::AddAutoAr(ss, y = y, lags = 10)}, ...)
```
Comment (nikosbosse) [inline R/forecast_cases.R:23]: add fable example

File: R/fable_model.R (hunk 2 above)
Comment (nikosbosse) [reply R/fable_model.R:34]: ah ok I see the date isn't returned anyway... Forget about my comment :)

File: R/forecast_rt.R [hunk 4]
```diff
@@ -21,7 +18,10 @@
 #'
 #' @examples
 #' forecast_rt(EpiSoon::example_obs_rts[1:10, ],
-#'             model = function(ss, y){bsts::AddAutoAr(ss, y = y, lags = 10)},
+#'             model = function(...){
+#'             EpiSoon::bsts_model(model = function(ss, y){
+#'             bsts::AddAutoAr(ss, y = y, lags = 10)}, ...)
+#'             },
```
Comment (nikosbosse) [inline R/forecast_rt.R:24]: fable example would be good

File: R/iterative_case_forecast.R [hunk 5]
```diff
@@ -12,7 +12,8 @@
 #' ## Iterative Rt forecast
 #' it_forecast <-
 #'   iterative_rt_forecast(EpiSoon::example_obs_rts,
-#'                         model = function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)},
+#'                         model = function(...) {EpiSoon::bsts_model(model =
+#'                     function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)}, ...)},
```
Comment (nikosbosse) [inline R/iterative_case_forecast.R:16]: inconsistent spacing, 
maybe fable example

File: R/iterative_rt_forecast.R [hunk 6]
```diff
@@ -12,7 +12,8 @@
 #'
 #'
 #' iterative_rt_forecast(EpiSoon::example_obs_rts,
-#'                       model = function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)},
+#'                       model = function(...) {EpiSoon::bsts_model(model =
+#'                     function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)}, ...)},
```
Comment (nikosbosse) [inline R/iterative_rt_forecast.R:16]: inconsistent spacing,
maybe fable example

File: R/plot_forecast.R [hunk 7]
```diff
@@ -9,18 +9,19 @@
...
 #' @return A `ggplot2` object
 #' @export
 #'
 #' @examples
 #'
 #' ## Forecast an Rt sample
 #' samples <- forecast_rt(EpiSoon::example_obs_rts[1:10, ],
-#'                      model = function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)},
+#'                      model = function(...) {EpiSoon::bsts_model(model =
+#'                     function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)}, ...)},
```
Comment (nikosbosse) [inline R/plot_forecast.R:24]: spacing

File: R/score_forecast.R [hunk 8]
```diff
@@ -17,7 +17,8 @@
 #'
 #' ## Fit a model (using a subset of observations)
 #' samples <- forecast_rt(EpiSoon::example_obs_rts[1:10, ],
-#'                      model = function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)},
+#'                      model = function(...) {EpiSoon::bsts_model(model =
+#'                     function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)}, ...)},
```
Comment (nikosbosse) [inline R/score_forecast.R:21]: spacing

File: R/summarise_case_forecast.R [hunk 9]
```diff
@@ -11,9 +11,8 @@
 #'
 #' ## Example forecast
 #' forecast <- forecast_rt(EpiSoon::example_obs_rts,
-#'                         model = function(ss, y) {
-#'                         bsts::AddAutoAr(ss, y = y, lags = 7)
-#'                         },
+#'                         model = function(...) {EpiSoon::bsts_model(model =
+#'                     function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)}, ...)},
```
Comment (nikosbosse) [inline R/summarise_case_forecast.R:15]: spacing

File: R/summarise_forecast.R [hunk 10]
```diff
@@ -12,7 +12,8 @@
 #'
 #'
 #' samples <- forecast_rt(example_obs_rts,
-#'                        model = function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)},
+#'                        model = function(...) {EpiSoon::bsts_model(model =
+#'                     function(ss, y){bsts::AddSemilocalLinearTrend(ss, y = y)}, ...)},
```
Comment (nikosbosse) [inline R/summarise_forecast.R:16]: spacing

File: R/bsts_model.R (hunk 1 above)
Comment (nikosbosse) [reply R/bsts_model.R:20]: okk :)

**Pull request merged**